### DISCUSSION TITLE
Using conditional expressions in properties.

### DISCUSSION THREAD

**Discussion by drewhk** [slint-ui/slint#562] [Q&A] [answered]
Hi, I have a simple question. I would like to set the background color of a tile according to some condition. Think about a chessboard where every second tile is white. However, I do not want to hardcode color values in the logic/model, only in the widget. What I am after is something like this:

```
background: if (white) #eeeeec else #e9b96e;
```

The docs have two sections, one about conditional _statements_ and the other about conditional _elements_, but none of them is exactly what I want here. Is this feature there just not in the docs, or is this something missing?

Thank you for your help!

**Comment by ogoffart** [comment, marked as answer]
We support the ternary operator like in C or JS:
```
background : white ? #eeeeec : #e9b96e;
```

I realize this is badly documented, so i'll improve on that.

**Reply by drewhk** [reply]
Thanks I should have tried that, duh...
